### PR TITLE
draft release notes for 2.1.0

### PR DESCRIPTION
**Pull request by andreaturli**

### CODE REVIEW

**Comment by jclouds-commentator** [comment]
Go to http://4df7ef4a7c101a5d11f9-a246c0c5592cbc40541c6d2ceeb6e29f.r71.cf5.rackcdn.com/ to review your changes.

File: releasenotes/2.1.0.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+---
+layout: releasenotes
+title: Release Notes for Version 2.0.0
+date: 2016-11-15 12:00:00
```
Comment (nacx) [inline releasenotes/2.1.0.md]: Put the release date here. It should be the same than the publish date for this page.

File: releasenotes/2.1.0.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+---
+layout: releasenotes
+title: Release Notes for Version 2.0.0
```
Comment (nacx) [inline releasenotes/2.1.0.md]: 2.1.0

File: releasenotes/2.1.0.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+date: 2016-11-15 12:00:00
+release_notes: true
+permalink: /releasenotes/2.0.0/
```
Comment (nacx) [inline releasenotes/2.1.0.md]: 2.1.0

File: releasenotes/2.1.0.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+1. [Release Highlights](#highlights)
+1. [Compute Improvements](#computeimprovements)
+1. [BlobStore Improvements](#blobstoreimprovements)
+1. [Known Issues](#knownissues)
+1. [Credits](#credits)
+1. [Test Results](#test)
+
+## <a id="intro"></a>Introduction
+
+Apache jclouds 2.0 represents a significant milestone for the project. We are proud to support all major cloud providers in the marketplace with a mature, stable, codebase that is ready for production. You can [read the official announcement here](https://s.apache.org/NbYK). To get started with jclouds, please see the [jclouds installation guide](/start/install/).
```
Comment (nacx) [inline releasenotes/2.1.0.md]: 2.1

Comment (nacx) [reply releasenotes/2.1.0.md]: Also, if you put a placeholder for the link such as `https://s.apache.org/jclouds210` you can then generate a link for the announcement using that name and you won't have to edit the post later.

**Comment by jclouds-commentator** [comment]
Go to http://52f9b347c6aa8eb35408-326ce8443d102bf8173ebf3c7ac89da1.r37.cf5.rackcdn.com/ to review your changes.

**Comment by jclouds-commentator** [comment]
Go to http://ef76db5df87c220d0560-035f0ba595d77b703bd7f007c024956a.r42.cf5.rackcdn.com/ to review your changes.

**Comment by andreaturli** [comment]
thanks @nacx not sure why you were seeing all those references to 2.0.0 but I think they are now ok

**Comment by jclouds-commentator** [comment]
Go to http://04800a27d2b3ab8c59e7-9346c06763f734e0e32f5e6df94243aa.r58.cf5.rackcdn.com/ to review your changes.

File: releasenotes/2.1.0.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+## <a id="intro"></a>Introduction
+
+Apache jclouds 2.1.0 represents an important milestone for the project. We are proud to support all major cloud providers in the marketplace with a mature, stable, codebase that is ready for production. You can [read the official announcement here](https://s.apache.org/jclouds210). To get started with jclouds, please see the [jclouds installation guide](/start/install/).
+
+## <a id="highlights"></a>Release Highlights
+
+Apache jclouds 2.1.0 features include:
+
+* Modernize Openstack support
```
Comment (nacx) [inline releasenotes/2.1.0.md]: here I'd say "up to OpenStack Pike", which is the latest OpenStack version and it work fine with 2.1.0.

File: releasenotes/2.1.0.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+* Better support for AWS EC2 in terms of network capabilities and regions
+* Graduate Google Cloud Storage (GCS) and Backblaze B2 to core
+* Better integration with OSGi and Apache Karaf.
+* Numerous bug fixes and performance improvements.
+
+The complete list of fixed issues and improvements can be found [here](https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12314430&version=12334714).
+
+## <a id="computeimprovements"></a>Compute Improvements
+
+* Add Support OpenStack Keystone v3 API
```
Comment (nacx) [inline releasenotes/2.1.0.md]: Link to the Keystone V3 blog post.

File: releasenotes/2.1.0.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+* Promote Openstack Neutron to core
+* Better support of Openstack Neutron from Openstack Nova
```
Comment (nacx) [inline releasenotes/2.1.0.md]: Link to the context linking blog post (I have to finish it).

File: releasenotes/2.1.0.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+* Add support for Dimension Data
+* Add support Azure ARM Managed Disk
```
Comment (nacx) [inline releasenotes/2.1.0.md]: I think this was already present in 2.0.3

File: releasenotes/2.1.0.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+* [Jim Micheal Spring](https://twitter.com/jmspring) for his contributions to the Microsoft Azure ARM provider ant the OAuth APIs.
+* [Geoff Macartney](https://github.com/geomacy) for all his contributions to the Openstack and AWS providers.
+* [William Chen](https://github.com/swaqos) for his help with SoftLayer
+* Apache jclouds [PMC](http://people.apache.org/committers-by-project.html#jclouds-pmc) and community for verifying the release, particularly to [Ignasi Barrera](https://github.com/nacx) for all his help with Openstack and Azure ARM and the project management, in general.
+* Check out who has been busy on [Open Hub](https://www.openhub.net/p/jclouds/contributors?query=&sort=latest_commit).
+
+## <a id="test"></a>Test Results
+
+Please see the discussion and the vote threads for live test results for 2.0.0:
```
Comment (nacx) [inline releasenotes/2.1.0.md]: 2.1.0

File: releasenotes/2.1.0.md (hunk 8 above)
Comment (andreaturli) [reply releasenotes/2.1.0.md]: thx

File: releasenotes/2.1.0.md (hunk 9 above)
Comment (andreaturli) [reply releasenotes/2.1.0.md]: thx

File: releasenotes/2.1.0.md (hunk 5 above)
Comment (andreaturli) [reply releasenotes/2.1.0.md]: done

File: releasenotes/2.1.0.md (hunk 6 above)
Comment (andreaturli) [reply releasenotes/2.1.0.md]: good idea, done

File: releasenotes/2.1.0.md (hunk 7 above)
Comment (andreaturli) [reply releasenotes/2.1.0.md]: good idea. FYI I've put a placeholder `/blog/2018/01/16/keystone-v3/` for it

**Comment by jclouds-commentator** [comment]
Go to http://8bca845c0e48a6b7de0d-642c23479099d0ca574364957f7a2999.r93.cf5.rackcdn.com/ to review your changes.

File: releasenotes/2.1.0.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+* Improve OAuth capabilities adding filter customization per request
+* Update ProfitBricks to the latest API version
+* Add InternetGatewayApi and RouteTableApi to AWS EC2
+* Add support for R4 instance types in AWS EC2
+* Add Paris region to AWS EC2
+* Add Network API set feature add to SoftLayer provider
+
+## <a id="blobstoreimprovements"></a>BlobStore Improvements
+
+* Add Atmos time-limited signed URLs
```
Comment (gaul) [inline releasenotes/2.1.0.md]: Remove this, just generally mention improved Atmos compatibility.

File: releasenotes/2.1.0.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+## <a id="highlights"></a>Release Highlights
+
+Apache jclouds 2.1.0 features include:
+
+* Modernize Openstack support, up to Openstack Pike. See also [keystone v3](/blog/2018/01/16/keystone-v3/) and [Openstack Nova and Neutron](/blog/2018/02/16/nova-neutron/) blog posts
+* Graduate Microsoft Azure Resource Manager to core
+* Better support for AWS EC2 in terms of network capabilities and regions
+* Graduate Google Cloud Storage (GCS) and Backblaze B2 to core
+* Better integration with OSGi and Apache Karaf.
+* Numerous bug fixes and performance improvements.
```
Comment (gaul) [inline releasenotes/2.1.0.md]: Add:

* Compatible with Java 9 with `--add-modules java.xml.bind`
* Remove Clojure bindings
* Upgraded Guava to version 18

File: releasenotes/2.1.0.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+* Add InternetGatewayApi and RouteTableApi to AWS EC2
+* Add support for R4 instance types in AWS EC2
+* Add Paris region to AWS EC2
+* Add Network API set feature add to SoftLayer provider
+
+## <a id="blobstoreimprovements"></a>BlobStore Improvements
+
+* Add Atmos time-limited signed URLs
+* Support GCS temporary signed URLs
+* Add Azure Cool Blob Storage
```
Comment (gaul) [inline releasenotes/2.1.0.md]: Add:

* added storage tiers to all providers and portable abstraction
* allow larger Azure uploads
* improve multipart upload performance

**Comment by andreaturli** [comment]
thanks @gaul I've updated the release notes with your suggestions

**Comment by jclouds-commentator** [comment]
Go to http://7304807128365f8355d8-ecc0d6a32b2f448724f097767e53bdc9.r57.cf5.rackcdn.com/ to review your changes.

**Comment by andreaturli** [comment]
are we happy with that?

File: releasenotes/2.1.0.md [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+1. [Credits](#credits)
+1. [Test Results](#test)
+
+## <a id="intro"></a>Introduction
+
+Apache jclouds 2.1.0 represents an important milestone for the project. We are proud to support all major cloud providers in the marketplace with a mature, stable, codebase that is ready for production. You can [read the official announcement here](https://s.apache.org/jclouds210). To get started with jclouds, please see the [jclouds installation guide](/start/install/).
+
+## <a id="highlights"></a>Release Highlights
+
+Apache jclouds 2.1.0 features include:
```
Comment (demobox) [inline releasenotes/2.1.0.md]: [minor] "2.1.0 highlights include"? The below are not really "features"?

File: releasenotes/2.1.0.md [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+
+* Modernize Openstack support, up to Openstack Pike. See also [keystone v3](/blog/2018/01/16/keystone-v3/) and [Openstack Nova and Neutron](/blog/2018/02/16/nova-neutron/) blog posts
```
Comment (demobox) [inline releasenotes/2.1.0.md]: [minor] Add `.` at the end of bullet points for consistency?

Comment (demobox) [inline releasenotes/2.1.0.md]: [minor] "OpenStack" (capital 'S')

File: releasenotes/2.1.0.md [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+
+The complete list of fixed issues and improvements can be found [here](https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12314430&version=12334714).
+
+## <a id="computeimprovements"></a>Compute Improvements
+
+* Add Support OpenStack Keystone v3 API
+* Promote Openstack Neutron to core
+* Better support of Openstack Neutron from Openstack Nova
+* Add support for Dimension Data
+* Add Azure Virtual Machine Scale Set Support
```
Comment (demobox) [inline releasenotes/2.1.0.md]: [minor] "support" (lowercase 's')

File: releasenotes/2.1.0.md [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+* Add Azure Cool Blob Storage
+* Add storage tiers to all providers and portable abstraction
+* Allow larger Azure uploads
+* Improve multipart upload performance
+
+## <a id="credits"></a>Credits
+
+jclouds would like to thank everyone who contributed time and effort in order to make this release happen:
+
+* [Jim Micheal Spring](https://twitter.com/jmspring) for his contributions to the Microsoft Azure ARM provider ant the OAuth APIs.
```
Comment (demobox) [inline releasenotes/2.1.0.md]: [minor] "and the"

File: releasenotes/2.1.0.md [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+* [Geoff Macartney](https://github.com/geomacy) for all his contributions to the Openstack and AWS providers.
```
Comment (demobox) [inline releasenotes/2.1.0.md]: "OpenStack"

File: releasenotes/2.1.0.md [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+
+Apache jclouds 2.1.0 features include:
+
+* Modernize Openstack support, up to Openstack Pike. See also [keystone v3](/blog/2018/01/16/keystone-v3/) and [Openstack Nova and Neutron](/blog/2018/02/16/nova-neutron/) blog posts
+* Graduate Microsoft Azure Resource Manager to core
+* Better support for AWS EC2 in terms of network capabilities and regions
+* Graduate Google Cloud Storage (GCS) and Backblaze B2 to core
+* Better integration with OSGi and Apache Karaf.
+* Numerous bug fixes and performance improvements.
+* Compatible with Java 9 with --add-modules java.xml.bind
```
Comment (demobox) [inline releasenotes/2.1.0.md]: [minor] Put `--add-modules ...` in backticks, i.e. code font?

File: releasenotes/2.1.0.md [hunk 19]
```diff
@@ -0,0 +1,78 @@
...
+* Remove Clojure bindings
+* Upgraded Guava to version 18
+
+The complete list of fixed issues and improvements can be found [here](https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12314430&version=12334714).
+
+## <a id="computeimprovements"></a>Compute Improvements
+
+* Add Support OpenStack Keystone v3 API
+* Promote Openstack Neutron to core
```
Comment (demobox) [inline releasenotes/2.1.0.md]: "OpenStack" (and in next bullet point)

File: releasenotes/2.1.0.md [hunk 20]
```diff
@@ -0,0 +1,78 @@
...
+* Allow larger Azure uploads
+* Improve multipart upload performance
+
+## <a id="credits"></a>Credits
+
+jclouds would like to thank everyone who contributed time and effort in order to make this release happen:
+
+* [Jim Micheal Spring](https://twitter.com/jmspring) for his contributions to the Microsoft Azure ARM provider ant the OAuth APIs.
+* [Geoff Macartney](https://github.com/geomacy) for all his contributions to the Openstack and AWS providers.
+* [William Chen](https://github.com/swaqos) for his help with SoftLayer
```
Comment (demobox) [inline releasenotes/2.1.0.md]: [minor] End with `.` for consistency?

File: releasenotes/2.1.0.md [hunk 21]
```diff
@@ -0,0 +1,78 @@
...
+* Apache jclouds [PMC](http://people.apache.org/committers-by-project.html#jclouds-pmc) and community for verifying the release, particularly to [Ignasi Barrera](https://github.com/nacx) for all his help with Openstack and Azure ARM and the project management, in general.
```
Comment (demobox) [inline releasenotes/2.1.0.md]: [minor] "OpenStack"

File: releasenotes/2.1.0.md [hunk 22]
```diff
@@ -0,0 +1,78 @@
...
+* Check out who has been busy on [Open Hub](https://www.openhub.net/p/jclouds/contributors?query=&sort=latest_commit).
+
+## <a id="test"></a>Test Results
+
+Please see the discussion and the vote threads for live test results for 2.1.0:
+
+* RC1 [vote](https://lists.apache.org/thread.html/204722e3f0fefc55e62f235a566c608733d16c0ecf442284f1d9e989@%3Cdev.jclouds.apache.org%3E) and [discussion](https://lists.apache.org/thread.html/204722e3f0fefc55e62f235a566c608733d16c0ecf442284f1d9e989@%3Cdev.jclouds.apache.org%3E) threads.
```
Comment (demobox) [inline releasenotes/2.1.0.md]: Both links point to the `[VOTE]` thread?

File: releasenotes/2.1.0.md [hunk 23]
```diff
@@ -0,0 +1,78 @@
...
+* RC2 [vote](https://lists.apache.org/thread.html/2d2907fb83571f5890967a8c4c2801bc1bbb1ebc334fd9391f3ef752@%3Cdev.jclouds.apache.org%3E) and [discussion](https://lists.apache.org/thread.html/6ef19fe67bff98dc7afe1aad914a40e9373a664a88f41fb187f531df@%3Cdev.jclouds.apache.org%3E) threads.
```
Comment (demobox) [inline releasenotes/2.1.0.md]: Do we need the RC1 and RC2 links?

/cc @nacx

File: releasenotes/index.md [hunk 24]
```diff
@@ -6,6 +6,7 @@ permalink: /releasenotes/
 
 The Release Notes and Javadocs for every major release of Apache jclouds are listed chronologically below.
 
+* [2.1.0](/releasenotes/2.1.0) | [Javadoc](/reference/javadoc/2.1.x/)
```
Comment (demobox) [inline releasenotes/index.md]: This link seems to be broken:

http://jclouds.apache.org/reference/javadoc/2.1.x/

Is that expected?

Comment (andreaturli) [reply releasenotes/index.md]: yes @demobox I think I've not executed that script. When should I do that?

**Comment by andreaturli** [comment]
thanks @demobox for your review!

**Comment by nacx** [comment]
Staging site: http://7cf94f307dd1bb8cf1fa-8f1fba471c11f5574dc0107a5cd29139.r9.cf5.rackcdn.com/

File: releasenotes/index.md (hunk 24 above)
Comment (demobox) [reply releasenotes/index.md]: The steps are [here](https://cwiki.apache.org/confluence/display/JCLOUDS/Publishing+Javadoc), from what I can make out - listed as step 19 in the [release guide](https://cwiki.apache.org/confluence/display/JCLOUDS/Releasing+jclouds), I think?

Comment (andreaturli) [reply releasenotes/index.md]: @demobox I think you are right, I missed step 19

I think we need to update the notes at https://cwiki.apache.org/confluence/display/JCLOUDS/Publishing+Javadoc as they are not working for me. I've tried to tweak the script - see new commit in this PR - but I'm afraid I don't know enough of the process to fix it properly. I don't get the svn part, in particular

Comment (demobox) [reply releasenotes/index.md]: > but I'm afraid I don't know enough of the process to fix it properly

@nacx I think you've run this most recently...any thoughts?

> I don't get the svn section at line 55 shows me of course

From what I can see in [`deploy-site.sh`](https://github.com/jclouds/jclouds-site/blob/master/deploy-site.sh#L21-L25), I think the assumption is that that is run first. That would make the directory into which `jclouds-site` is cloned both a Git and an SVN repository, and then the [`svn status` command](https://github.com/jclouds/jclouds-site/blob/master/deploy-javadoc.sh#L54) in `deploy-javadoc.sh` should work..?

Comment (nacx) [reply releasenotes/index.md]: Javadocs are not pushed to the staging site; we just push them once the page is live, so I'd say there's no issue here.

**Comment by andreaturli** [comment]
is it ok to merge @nacx ?

**Comment by nacx** [comment]
Can you push the rebased branch so we can see it after-rebase?
LGTM, btw

**Comment by jclouds-commentator** [comment]
Go to http://83f7696a41fdd0d9c3a3-7c6167cff13c42232b59a50f18033ee5.r93.cf5.rackcdn.com/ to review your changes.

File: releasenotes/2.1.0.md [hunk 25]
```diff
@@ -0,0 +1,78 @@
...
+1. [Introduction](#intro)
+1. [Release Highlights](#highlights)
+1. [Compute Improvements](#computeimprovements)
+1. [BlobStore Improvements](#blobstoreimprovements)
+1. [Credits](#credits)
+1. [Test Results](#test)
+
+## <a id="intro"></a>Introduction
+
+Apache jclouds 2.1.0 represents an important milestone for the project. We are proud to support all major cloud providers in the marketplace with a mature, stable, codebase that is ready for production. You can [read the official announcement here](https://s.apache.org/jclouds210). To get started with jclouds, please see the [jclouds installation guide](/start/install/).
```
Comment (nacx) [inline releasenotes/2.1.0.md]: The link to the announcement thread is not correct.

File: releasenotes/2.1.0.md [hunk 26]
```diff
@@ -0,0 +1,78 @@
...
+* [Geoff Macartney](https://github.com/geomacy) for all his contributions to the OpenStack and AWS providers.
+* [William Chen](https://github.com/swaqos) for his help with SoftLayer.
+* Apache jclouds [PMC](http://people.apache.org/committers-by-project.html#jclouds-pmc) and community for verifying the release, particularly to [Ignasi Barrera](https://github.com/nacx) for all his help with OpenStack and Azure ARM and the project management, in general.
+* Check out who has been busy on [Open Hub](https://www.openhub.net/p/jclouds/contributors?query=&sort=latest_commit).
+
+## <a id="test"></a>Test Results
+
+Please see the discussion and the vote threads for live test results for 2.1.0:
+
+* RC1 [vote](https://lists.apache.org/thread.html/204722e3f0fefc55e62f235a566c608733d16c0ecf442284f1d9e989@%3Cdev.jclouds.apache.org%3E) and [discussion](https://lists.apache.org/thread.html/42711705a1b784a23b2192a60642ce1abc45ec7b9a74ce1e6553d1c4@%3Cdev.jclouds.apache.org%3E) threads.
```
Comment (nacx) [inline releasenotes/2.1.0.md]: Discuss link for RC1 points to 2.0.3.

File: releasenotes/2.1.0.md (hunk 25 above)
Comment (andreaturli) [reply releasenotes/2.1.0.md]: thanks, fixed

File: releasenotes/2.1.0.md (hunk 26 above)
Comment (andreaturli) [reply releasenotes/2.1.0.md]: thx, fixed

**Comment by jclouds-commentator** [comment]
Go to http://89c2ad2c745da05a9c62-9548ec5d02bc4c8f8745ef087a9c4d67.r83.cf5.rackcdn.com/ to review your changes.

**Comment by jclouds-commentator** [comment]
Go to http://0e33b5b22d14ffa07427-41f9b08bff70d1d1332761fd399f50bb.r47.cf5.rackcdn.com/ to review your changes.

**Comment by andreaturli** [comment]
merged at [master](http://git-wip-us.apache.org/repos/asf/jclouds-site/commit/314f3b56)

File: releasenotes/2.1.0.md (hunk 25 above)
Comment (demobox) [reply releasenotes/2.1.0.md]: > The link to the announcement thread is not correct.

Thanks for catching that, Ignasi!

Comment (andreaturli) [reply releasenotes/2.1.0.md]: Is it ok now?

**Pull request closed without merging**